src/coreComponents/fileIO/timeHistory/HistoryCollectionBase.cpp

```diff
@@ -84,108 +84,132 @@ bool HistoryCollectionBase::execute( real64 const time_n,
+          ++targetTokenLength;
+          targetTokens.insert( targetTokens.begin()+1, singleMeshLevelName );
+        }
+        else
+        {
+          bool levelFound = false;
+          meshBody.forMeshLevels( [&]( MeshLevel const & meshLevel )
+          {
+            if( meshLevel.getName()==targetTokens[1] )
+            {
+              levelFound=true;
+            }
+          } );
+
+          GEOSX_ERROR_IF( !levelFound,
+                          GEOSX_FMT( "MeshLevel ({}) is specified, but not found.",
+                                     targetTokens[1] ) );
+        }
+      }
+      else if( !meshBody.getMeshLevels().hasGroup< MeshLevel >( targetTokens[1] ) )
+      {
+        //GEOSX_LOG_RANK_0( "In TimeHistoryCollection.hpp, Mesh Level Discretization not specified, "
+        //                  "using baseDiscretizationString()." );
+
+        string const baseMeshLevelName = MeshBody::groupStructKeys::baseDiscretizationString();
+        ++targetTokenLength;
+        targetTokens.insert( targetTokens.begin()+1, baseMeshLevelName );
+      }
+
+      string meshLevelName = targetTokens[1];
+      MeshLevel const & meshLevel = meshBody.getMeshLevel( meshLevelName );
+      targetGroup = &meshLevel;
+
+
+      if( targetTokens[2]== MeshLevel::groupStructKeys::elemManagerString() )
+      {
+        ElementRegionManager const & elemRegionManager = meshLevel.getElemManager();
+        string const elemRegionName = targetTokens[3];
+        ElementRegionBase const & elemRegion = elemRegionManager.getRegion( elemRegionName );
+        string const elemSubRegionName = targetTokens[4];
+        ElementSubRegionBase const & elemSubRegion = elemRegion.getSubRegion( elemSubRegionName );
+        targetGroup = &elemSubRegion;
+      }
+      else
+      {
+        for( localIndex pathLevel = 2; pathLevel < targetTokenLength; ++pathLevel )
+        {
+          dataRepository::Group const * const childGroup = targetGroup->getGroupPointer( targetTokens[pathLevel] );
+          if( childGroup != nullptr )
+          {
+            targetGroup=childGroup;
+          }
+          else
+          {
+            GEOSX_THROW( targetTokens[pathLevel] << " not found in path " <<
+                         objectPath << std::endl <<
+                         "Children available in " <<
+                         stringutilities::join( targetTokens.begin(),
+                                                targetTokens.begin()+pathLevel,
+                                                '/' ) <<
+                         ": " << targetGroup->dumpChildrenName(),
+                         std::domain_error );
+          }
+        }
       }
+      return targetGroup;
     }
-    return targetGroup;
+  }
+  catch( std::domain_error const & e )
+  {
+    throw std::domain_error( "\n***** ERROR: " + getName() +
+                             " has a wrong objectPath. The following error have been thrown:" +
+                             e.what() );
   }
```

Comment:
Why do you catch to resend the same (or nearly the same) exception?

Reply:
In order to add context to this exception, here is an example of this exception message:
```
***** ERROR: initialPressure has a wrong objectPath. The following error have been thrown:
***** LOCATION: /appli_PITSI/MelvinRey/GEOSX_repos/GEOSX/src/coreComponents/mesh/MeshObjectPath.cpp:225
***** Controlling expression (should be false): !foundMatch
***** Rank 0: elementRegionsGroup doesn't have a child named badRegion.
elementRegionsGroup have the following children: reservoir, wellRegion
```
This catch adds the first line, it helps to know from which XML attribute of which group the exception comes from. If the name isn't mentioned, it can be hard for a user retrieve from which Group this exception comes from.

Reply:
When you throw the initial exception, `getName` is already available, isn't it?
Can't you furbish the message once for all?

Reply:
The original exception is thrown from a `MeshObjectPath`, as it doesn't know which `Group` is using it to find an object, I don't think I can call `getName()` from there.

Do you think this would be cleaner ?
```
***** ERROR
***** LOCATION: /appli_PITSI/MelvinRey/GEOSX_repos/GEOSX/src/coreComponents/mesh/MeshObjectPath.cpp:225
***** Controlling expression (should be false): !foundMatch
***** initialPressure has a wrong objectPath.
elementRegionsGroup doesn't have a child named badRegion.
elementRegionsGroup have the following children: reservoir, wellRegion
```
Or are you talking about the GEOSX_THROW message ?

Reply:
OK; most likely there are multiple throws that I did not detect. (that's the problem of exceptions...).
Forget about this thread then!